applications/HDF5Application/custom_utilities/container_io_utils.h

```diff
@@ -0,0 +1,477 @@
+
+    template<class TDataType>
+    using ComponentDataType = TDataType;
+
+    ///@}
+    ///@name Life cycle
+    ///@{
+
+    HistoricalIO(const unsigned int StepIndex) : mStepIndex(StepIndex) {}
+
+    ///@}
+    ///@name Public operations
+    ///@{
+
+    template<class TDataType>
+    inline const TDataType& GetValue(
+        const ModelPart::NodeType& rNode,
+        const Variable<TDataType>& rVariable,
+        char& rTLS) const
+    {
+        return rNode.FastGetSolutionStepValue(rVariable, mStepIndex);
+    }
+
+    template<class TDataType>
+    inline void SetValue(
+        ModelPart::NodeType& rNode,
+        const Variable<TDataType>& rVariable,
+        const TDataType& rValue) const
+    {
+        rNode.FastGetSolutionStepValue(rVariable, mStepIndex) = rValue;
+    }
+
+    ///@}
+
+private:
+    ///@name Private member variables
+    ///@{
+
+    const unsigned int mStepIndex;
+
+    ///@}
+};
+
+class NonHistoricalIO
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    template<class TDataType>
+    using TLSType = char;
+
+    template<class TDataType>
+    using ComponentDataType = TDataType;
+
+    ///@}
+    ///@name Public operations
+    ///@{
+
+    template<class TEntityType, class TDataType>
+    inline const TDataType& GetValue(
+        const TEntityType& rEntity,
+        const Variable<TDataType>& rVariable,
+        char& rTLS) const
+    {
+        return rEntity.GetValue(rVariable);
+    }
+
+    template<class TEntityType, class TDataType>
+    inline void SetValue(
+        TEntityType& rEntity,
+        const Variable<TDataType>& rVariable,
+        const TDataType& rValue) const
+    {
+        rEntity.SetValue(rVariable, rValue);
+    }
+
+    ///@}
+};
+
+class BossakIO
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    template<class TDataType>
+    using TLSType = char;
+
+    template<class TDataType>
+    using ComponentDataType = TDataType;
+
+    ///@}
+    ///@name Life cycle
+    ///@{
+
+    BossakIO(const double AlphaBossak) : mAlphaBossak(AlphaBossak) {}
+
+    ///@}
+    ///@name Public operations
+    ///@{
+
+    template<class TDataType>
+    inline TDataType GetValue(
+        const ModelPart::NodeType& rNode,
+        const Variable<TDataType>& rVariable,
+        char& rTLS) const
+    {
+        if (rVariable == ACCELERATION) {
+            return (1.0 - mAlphaBossak) * rNode.FastGetSolutionStepValue(rVariable, 0) + mAlphaBossak * rNode.FastGetSolutionStepValue(rVariable, 1);
+        } else {
+            return rNode.FastGetSolutionStepValue(rVariable);
+        }
+    }
+
+    template<class TDataType>
+    inline void SetValue(
+        ModelPart::NodeType& rNode,
+        const Variable<TDataType>& rVariable,
+        const TDataType& rValue) const
+    {
+        rNode.FastGetSolutionStepValue(rVariable) =  rValue;
+    }
+
+    ///@}
+
+private:
+    ///@name Private member variables
+    ///@{
+
+    const double mAlphaBossak;
+
+    ///@}
+};
```

Comment:
The way transient adjoints were done originally by using this modified output scheme of adjoints because, at the early days of transient adjoints, there were so little room for transient data passing. I would keep this for time being, until the old `VMSAdjointElement` is retired. [It is the only place this is required for fluid transient adjoint computations]

What this does is, it stores the adjoint specific `ACCELERATION` value as `ACCELERATION` in the HDF5 otuput, so when the old `VMSAdjointElement` reads `ACCELERATION`, it gets the adjoint sepcific `ACCELERATION` instead of the actual `ACCELERATION`. I understand this is workaround, we had in early days but for now, lets keep it.

Reply:
fine, just please add a `@todo` so we can get rid of it when we drop support for `VMSAdjointElement`